js/control/TrackAnalysis.js

```diff
@@ -180,13 +180,19 @@ BR.TrackAnalysis = L.Class.extend({
                         case 'surface':
                         case 'smoothness':
                             if (typeof analysis[tagName][wayTagParts[1]] === 'undefined') {
-                                let formattedName = i18next.t([
-                                    'sidebar.analysis.data.' + tagName + '.' + wayTagParts[1],
-                                    wayTagParts[1],
-                                ]);
+                                let formattedName;
+
                                 if (tagName.indexOf('maxspeed') === 0) {
-                                    formattedName += ' km/h';
+                                    formattedName = i18next.t('sidebar.analysis.data.maxspeed', {
+                                        maxspeed: wayTagParts[1],
```

Comment:
You could totally remove the `if` specific to maxSpeed, and instead have a fallback mecanism:

1. try to use the full key `sidebar.analysis.data.highway.footway`
2. if not available, try `sidebar.analysis.data.highway`
3. otherwise simply fallback to highway type eg `footway` here  

It should cover all cases while being agnostic of the kind of analysis (if there are more analysis coming in the future :smirk:)

```js
const highwayType = wayTagParts[1];
formattedName = i18next.t(
    [
         'sidebar.analysis.data.' + tagName + '.' + highwayType,
         'sidebar.analysis.data.' + tagName,
          highwayType
    ], 
    {highwayType: highwayType}
);
```

If a translation key needs the `highwayType` it can use it, otherwise it can just skip it and it should work just fine.

Reply:
I spotted some more possibilities to simplify the translations for the analysis sidebar, but I'd rather create a separate PR as there're changes on places unrelated to this PR. 

Can you agree with this?